### DISCUSSION TITLE
I want to detect presence of image in a page based on thickness of black lines

### DISCUSSION THREAD

**Discussion by srini8080** [ImageMagick/ImageMagick#6370] [Digital Image Processing]
Imagemagick 7+
Centos 8.1

I have 2 pages. One has  just text and the other has image+text. I used -auto-threshold otsu to make them into thick lines. 

first one has just text
second one has image and text
![x](https://github.com/ImageMagick/ImageMagick/assets/12046302/f4b0c421-8f9b-4be7-bca3-15e0cd44cbf3)
![x1](https://github.com/ImageMagick/ImageMagick/assets/12046302/6c37d932-f1ea-44c3-9205-d06f53541e6d)

Based on the thickness of black lines, how do I detect if the page has a image or just text?

Thanks

**Comment by fmw42** [comment]
What is the question?  Your image is just stripes. I see no text and no image.

**Comment by srini8080** [comment]
![aesopsfables00aeso-000291_final](https://github.com/ImageMagick/ImageMagick/assets/12046302/e5f9a039-6a6f-4e42-979f-188c24eec0a7)
![aesopsfables00aeso-000292_final](https://github.com/ImageMagick/ImageMagick/assets/12046302/58ea5e03-cecc-4b17-8789-73e0ed116ef1)

I want to detect that page 2 has an image and page 1 has just text.

I thought if I can bring them down to just thick dark lines and see if the line above a threshold value can be detected as a image.

Thanks

**Comment by fmw42** [comment]
Threshold the image. Average one column. Scale back to full size. Threshold so get stripes.  Then use connected components to get the bounding boxes of the black regions. Then loop over the bounding boxes. If the bounding box has ht > 60 (text size for double row starting with Cap letters), then crop the image using the bounding box.

NOTE:  We have to change the Internal Field Separator from a space to a new line before the connected components and then return it to a space afterwards. Otherwise, all the bounding boxes are treated as one string list rather than on separate lines.

Input:

![text](https://github.com/ImageMagick/ImageMagick/assets/13458218/b2f0ebe3-7147-496f-9bd3-efa456b25cbe)


Unix Syntax:


```
OLDIFS=$IFS
IFS=$'\n'
array=(`im7 magick text.jpg \
-set option:wh "%wx%h" \
-threshold 50% \
-scale 1x! -scale "%[wh]"! \
-negate -threshold 0% -negate \
-type bilevel \
-define connected-components:verbose=true \
-define connected-components:mean-color=true \
-connected-components 8 null: | \
grep "gray(0)" | awk '{print $2}'`)
IFS=$OLDIFS

num=${#array[*]}
j=1
for ((i=0; i<num; i++)); do
bbox="${array[$i]}"
ht=`echo $bbox | tr "x" "+" | cut -d+ -f2`
if [ $ht -gt 60 ]; then
im7 magick text.jpg -crop $bbox +repage text_crop_$j.jpg
j=$((j+1))
fi
done

```

![text_crop_1](https://github.com/ImageMagick/ImageMagick/assets/13458218/9a90731b-9e90-4ae2-932e-129442dbd480)

**Reply by srini8080** [reply]
Thanks. I made the 60 into 10% of the image height so that it can work with any image.
